File: c#/EmailMarketingApi/Locaweb.EmailMarketing.Api/Exemplos/ListarContatos.cs
/*
 *  Copyright (c) 2009, Locaweb LTDA.
 * 	Todos os direitor reservados.
 *
 *  Está é uma API exemplo que facilita a utilização dos web services do Email Marketing.
 *
 * versao 1.0
 * mais detalhes em http://wiki.locaweb.com.br/pt-br/APIs_do_Email_Marketing
 */
using System;
using System.Collections.Generic;
using Locaweb.EmailMarketing.Api.Contatos;
using System.Net;
using System.IO;

namespace Locaweb.EmailMarketing.Api.Exemplos
{
    class ListarContatos
    {
        static void Main(string[] args)
        {
            //para saber quais valores utilizar, consulte a página de configurações do Email Marketing
            //mais informações em http://wiki.locaweb.com.br/pt-br/APIs_do_Email_Marketing

            const string HOSTNAME       = "";
            const string LOGIN          = "";
            const string CHAVE_API      = "";
            const string HOSTNAME_SUFIX = "locaweb.com.br";            

            RepositorioContatos contatoApi = new RepositorioContatos(HOSTNAME, LOGIN, CHAVE_API, HOSTNAME_SUFIX);

            try
            {
                List<Contato> contatos;

                for (int pagina = 1; (contatos = contatoApi.obterValidos(pagina)) != null ; pagina++)
                {
                    Console.WriteLine("pagina " + pagina);
                    
                    foreach (Contato c in contatos)
                    {
                        Console.WriteLine(string.Format("nome:{0}, email:{1}, dataNasc:{2}, estado:{3}",
                            c.nome, c.email, c.dataDeNascimento, c.estado));
                    }
                }
                Console.WriteLine(System.GC.GetTotalMemory(true).ToString());

            }
            catch (EmktApiException e)
            {
                Console.WriteLine("Erro na chamada da API: " + e.Message);
            }
            catch (Exception e)
            {
                Console.WriteLine("Ocorreu um erro inesperado: " + e.Message);
            }
        }
    }
}

File: php/EmailMarketingApi/src/listarContatos.php
<?php
/**
 *  Copyright (c) 2009, Locaweb LTDA.
 * 	Todos os direitor reservados.
 *
 *  Este é um código exemplo de como utilizar a API do Email Marketing.
 *
 * @version 1.0
 * @see http://wiki.locaweb.com.br/pt-br/APIs_do_Email_Marketing
 */

	error_reporting(E_ALL);
	require_once dirname(__FILE__).'/RepositorioContatos.php';

	// Esses valores podem ser obtidos na página de configurações do Email Marketing
	$hostName = '';
	$login 		= '';
	$chaveApi = '';
	$repositorio 	= new RepositorioContatos($hostName, $login, $chaveApi);

	print "\ncontatos validos\n";

	try{
		for($pagina=1; $contatos = $repositorio->obterValidos($pagina); $pagina++) {
			foreach($contatos as $contato) {
				//caso necessite de outros campos do contato, utilizar o print_r($contato) para visualizar os campos disponiveis;
				print "email: {$contato->email}|nome: {$contato->nome}|" .
						"datanasc: {$contato->datadenascimento}| estado: {$contato->estado}\n";
			}
		}
	}catch(Exception $e){
		print "Erro ao chamar a API: {$e->getMessage()}\n";
		exit;
	}
?>
